.changeset/poor-boats-reflect.md

```diff
@@ -0,0 +1,5 @@
+---
+"stylelint": minor
+---
+
+Changed: `secondaryOptions` type
```

Comment:
[suggest] Minor. The following may be more explicit:

```suggestion
Changed: `secondaryOptions` argument type of the `Rule` function
```

Reply:
FYI, the `Rule` type is exposed:
https://github.com/stylelint/stylelint/blob/18dff4f65360b20e434c572850a24d88e035c614/types/stylelint/index.d.ts#L271-L274

Reply:
Yes but RuleBase is internal :)

```ts
	/** @internal */
	export type RuleBase<P = any, S = any> = (
```

It flows back to `Rule`, you are right.
I thought I would be spared on a technicality.